python/EMR_MASTER_NO_PUBLIC_IP/EMR_MASTER_NO_PUBLIC_IP.py

```diff
@@ -0,0 +1,498 @@
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+    evaluations = []
+
+    #Building service clients using get_client()
+
+    emr_client = get_client('emr', event)
+    ec2_client = get_client('ec2', event)
+    sts_client = get_client('sts', event)
+
+
+    #Listing RUNNING and WAITING clusters
+    cluster_list = listing_clusters(emr_client)
+
+    #If no RUNNING and WAITING clusters then return NOT_APPLICABLE
+    if not cluster_list:
+        return "NOT_APPLICABLE"
+
+    #Main logic for compliance deduction in cluster_dns_mapping()
+    cluster_list_valid, cluster_dns_list = cluster_dns_mapping(cluster_list, emr_client, ec2_client)
+
+    #cluster_dns_list is a list of lists which has cluster id and public DNS mappings
+    #If public DNS is empty then Compliant else not Compliant
+
+    if cluster_dns_list:
+        for cluster in cluster_dns_list:
+            print(cluster[0])
+            if cluster[1] == '':
+                evaluations.append(build_evaluation(cluster[0],
+                                                    'COMPLIANT',
+                                                    event,
+                                                    "AWS::EMR::Cluster"))
+            else:
+                evaluations.append(build_evaluation(cluster[0],
+                                                    'NON_COMPLIANT',
+                                                    event,
+                                                    "AWS::EMR::Cluster",
+                                                    annotation="The EMR Cluster's master has a public IP"))
+
+    #cluster_list_valid is a list of only cluster ids which have Compliant clusters
+    if cluster_list_valid:
+        for cluster in cluster_list_valid:
+            print(cluster)
+            evaluations.append(build_evaluation(cluster, 'COMPLIANT', event, "AWS::EMR::Cluster"))
```

Comment:
The ResourceType is already defined in `DEFAULT_RESOURCE_TYPE`. No need to add `"AWS::EMR::Cluster"`